string[] keys
float32[] values
---